configs/components/oifs/oifs.yaml

```diff
@@ -1044,10 +1059,11 @@ preprocess:
                 type: shell
 # postprocessing is not supported, and causes a crash in release 5 if not commented
 # seb-wahl 2020-11-27
-#postprocess:
-#        postprocess_shell:
-#                method: "${general.esm_namelist_dir}/../configs/components/oifs/oifs-43r3-postprocess.sh ${work_dir} ECE3 ${start_date!syear!smonth!sday} ${end_date!syear!smonth!sday}"
-#                type: shell
+postprocess:
+        postprocess_shell:
+                method: "${general.esm_namelist_dir}/../configs/components/oifs/oifs-43r3-postprocess.sh ${work_dir} ECE3 ${start_date!syear!smonth!sday} ${end_date!syear!smonth!sday}"
```

Comment:
This only works if you have a .esmtoolsrc file (which we actually abandoned with v5). That is why it is commented. I talk to the AWI people asap to see whether they have a solution. I've tried to solve it in the past, but it hasn't worked.

Reply:
Yes, we have a solution. After `.esmtoolsrc` becoming obsolete, I added some get/set functionality to the `esm_versions` command. Let's have a look at these examples:

$ esm_versions get esm_tools
```
esm_tools
---------
Version:         5.0.5
File Path:       /mnt/lustre01/pf/a/a271096/esm_all_packages/esm_tools
Branch:          hotfix/fix_lang_issues
Git Describe:    v5.0.8-8-g4ce568f
```

$ esm_versions get esm_tools **file_path**
`/mnt/lustre01/pf/a/a271096/esm_all_packages/esm_tools`

Let's use **command substitution** in shell
``ls `esm_versions get esm_tools file_path`/namelists/oifs``
or 
`ls $(esm_versions get esm_tools file_path)/namelists/oifs`

This yields
`40r1  43r3  namelist.oifs.esm.sh`

and 

`$ file $(esm_versions get esm_tools file_path)/configs/components/oifs/oifs-43r3-postprocess.sh`
gives 
`/some/long/path/esm_tools/configs/components/oifs/oifs-43r3-postprocess.sh: Bourne-Again shell script text executable`

I think this is what you are looking for. @seb-wahl: we can discuss that this morning at the meeting. Have a nice week!

Reply:
Hi @denizural @seb-wahl 

"general.esm_namelist_dir" occurs three times in oifs.yaml, and I replaced all of them with "$(esm_versions get esm_tools file_path)/namelists/". 

```bash
namelist_dir: "$(esm_versions get esm_tools file_path)/namelists/oifs/${version}/pictl/"
```

The preprocessing seems ok, but the namelist fort.4 can't be found. For example: 

```bash
(base) blogin1:~/esm_tools/runscripts/oifs $ esm_runscripts oifs-43r3-tco95-amip-blogin-test_JK.yaml -e oifs-test5-23 -c 
First year, checking if we need to compile...

================================================================================
STARTING SIMULATION JOB!
Experiment ID = oifs-test5-23
Setup = oifs
Experiment is installed in:
       /scratch/usr/shkjocke/esm-slask//oifs-test5-23
================================================================================

File not found: $(esm_versions get esm_tools file_path)/namelists/oifs/43r3/pictl/fort.4...
```

Seems that ```$(esm_versions get esm_tools file_path)``` is not expanded as expected. Can we solve this somehow?

Cheers
Joakim

Reply:
Sorry I think I put you on the wrong path. @denizural @joakimkjellsson I have a look at this now... Postprocessing does not work because there is no postprocessing plugin, I think. I let you know once I have a solution